fallback to initials if the image is not loaded (#40)

danielbprice opened the pull request:
Also, fix the version of vue-test-utils to 1.0.0-beta.20 to protect from breaking changes

saintplay commented:
Hi, interesting solution, did you take this pattern from somewhere?

danielbprice commented:
Yes, the technique is based on information I found here: https://stackoverflow.com/questions/5057990/how-can-i-check-if-a-background-image-is-loaded

danielbprice commented:
I think another possibility would be to use something like https://github.com/desandro/imagesloaded ; based on my very quick read of its code (and I am not an expert) it does something somewhat similar-- creating a `new Image()` and then attaching events to that, although it is has more features and is battle tested across multiple browsers.  It might be overkill for this application.  I definitely didn't know how you would feel about adding a dependency.

andreasvirkus commented:
@saintplay would you consider merging this?

saintplay commented:
I didn't have time to try this. @andreasvirkus can you confirm if this works?

danielbprice commented:
(I'm still here and happy to adjust this PR if there is review feedback)

andreasvirkus commented on documentation/_getting-started.pug:
```diff
@@ -257,6 +257,17 @@ a(href='https://github.com/eliep/vue-avatar')
+                &lt;!-- fallback to initials if image doesn't load --&gt;
+                &lt;avatar
+                  src="./static/luke-skywalker.png"
+                  :size="100"&gt;
```
Could you also add the `username` prop to the example here? Otherwise it's unclear where the `LS` initials come from in the example

andreasvirkus commented on package.json:
```diff
@@ -28,7 +28,7 @@
   "dependencies": {},
   "devDependencies": {
-    "@vue/test-utils": "^1.0.0-beta.20",
+    "@vue/test-utils": "=1.0.0-beta.20",
```
Why was this change necessay?

danielbprice replied:
Removed this from latest.

andreasvirkus commented:
@saintplay Checked it out and it seems to work great and not break any behaviour. You can see that the `luke-skywalker.png` is not found but the initials are still displayed.

![Screenshot 2020-05-15 at 11 27 48](https://user-images.githubusercontent.com/9140811/82029196-50ee9180-969f-11ea-865d-4b631dc052e5.png)

@danielbprice i've added a minor comment and a question to your PR, but otherwise really eager to see this merged 🎉

danielbprice commented:
I've removed the extraneous edit to package.json (I don't know why, but I can no longer reproduce the problem with vue-test that led me to do this; perhaps I hit a window where something was broken upstream which is now fixed) and fixed the issue with the documentation noted by @andreasvirkus 

Wasn't really sure if I was supposed to update the bundle, so let me know and I can strip out that change if not.

andreasvirkus commented:
LGTM @saintplay

saintplay commented:
@danielbprice it seems like you need to remove vue-avatar.min.js file or update your branch with `git rebase`

saintplay commented:
Sorry I'm late, I didn't get a notification.

Released in [v2.3.0](https://github.com/eliep/vue-avatar/releases/tag/2.3.0)

danielbprice commented:
(Belatedly) Thank you!  I appreciate it.
